lib/json_schema/validator.rb

```diff
@@ -506,7 +506,13 @@ def validate_type(schema, data, errors, path)
       if valid_types.any? { |t| data.is_a?(t) }
         true
       else
-        message = %{#{data.inspect} is not #{ErrorFormatter.to_list(schema.type)}.}
+        key = if schema.fragment =~ /patternProperties/
+                idx = schema.split_pointer.index("patternProperties")
+                schema.split_pointer[idx - 1]
```

Comment:
I'm kind of thinking that maybe we just move `split_pointer` to a local variable here. Thoughts?